File: Web/RMA-Web-Shell/projects/digicare/src/app/medical-form-manager/models/medical-report-form.ts
export class MedicalReportForm {
  medicalReportFormId: number;
  medicalReportSystemSource: number;
  consultationDate: Date;
  icd10Codes: string;
  icd10CodesJson: string;
  healthcareProviderName: string;
  healthcareProviderPracticeNumber: string;
  healthcareProviderId: number;
  workItemId: number;
  compcareMedicalReportId: number;
  reportCategoryId: number;
  reportCategoryData: string;
  reportTypeId: number;
  reportStatus: string;
  reportStatusDetail: string;
  reportStatusId: number;
  reportDetail: string;
  reportDate: Date;
  unfitStartDate: Date;
  unfitEndDate: Date;
  nextReviewDate: Date;
  documentStatusId: number;
  documentId: number;
  isDeleted:boolean;
  medicalReportRejectionReasonId : number
  medicalReportRejectionReasonDescription : string

  // Claim details
  claimReferenceNumber: string;
  eventCategoryId: number;
  personEventId: number;
  claimId: number;
  eventDate: Date;
  name: string;
  surname: string;
  gender: string;
  dateOfBirth: Date;
  contactNumber: string;
  industryNumber: string;
  employerName: string;
  claimantOccupation: string;
  isDeclarationAccepted: boolean;
  isICD10CodeMatch: boolean;

  // auditInfo
  createdBy: string;
  modifiedBy: string;
  tenantId: number;
  createdDate: Date;
  ModifiedDate: Date;

  isUnfitForWork(): boolean {
    return (this.unfitStartDate) && this.unfitStartDate > new Date();
  }
}

File: Service/Admin/RMA.Service.Admin.MasterDataManager.Contracts/Entities/MedicalReportForm.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;

namespace RMA.Service.Admin.MasterDataManager.Contracts.Entities
{
    public class MedicalReportForm : Integration
    {
        public int MedicalReportFormId { get; set; }
        public SourceSystemEnum MedicalReportSystemSource { get; set; }

        #region ClaimDetails
        public string ClaimReferenceNumber { get; set; }
        public int? ClaimId { get; set; }
        public int PersonEventId { get; set; }
        public int EventCategoryId { get; set; }
        public System.DateTime EventDate { get; set; }
        public System.DateTime DateOfBirth { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Gender { get; set; }
        public string ContactNumber { get; set; }
        public string IndustryNumber { get; set; }
        public string ClaimantOccupation { get; set; }
        public string EmployerName { get; set; }
        #endregion

        public DateTime ConsultationDate { get; set; }
        public DateTime ReportDate { get; set; }
        public string Icd10Codes { get; set; }
        public string Icd10CodesJson { get; set; }
        public string HealthcareProviderName { get; set; }
        public string HealthcareProviderPracticeNumber { get; set; }
        public int HealthcareProviderId { get; set; }
        public int WorkItemId { get; set; }
        public int CompcareMedicalReportId { get; set; }
        public int ReportCategoryId { get; set; }
        public int ReportTypeId { get; set; }
        public string ReportStatus { get; set; }
        public string ReportStatusDetail { get; set; }
        public int ReportStatusId { get; set; }
        public int? MedicalReportRejectionReasonId { get; set; }
        public string MedicalReportRejectionReasonDescription { get; set; }
        public int TenantId { get; set; }
        public bool IsDeclarationAccepted { get; set; }

        public DateTime? UnfitStartDate { get; set; }
        public DateTime? UnfitEndDate { get; set; }
        public string BodySideDetails { get; set; }
        public System.DateTime? NextReviewDate { get; set; }
        public int? DocumentId { get; set; }
        public string ReportCategoryData { get; set; }

        public DocumentStatusEnum? DocumentStatusId { get; set; }


        public bool IsUnfitForWork() { return UnfitStartDate.HasValue && UnfitStartDate.Value > DateTime.MinValue; }

        public int EstimatedDaysOff
        {
            get
            {
                if (UnfitStartDate.HasValue && UnfitEndDate.HasValue)
                    return (UnfitEndDate.Value - UnfitStartDate.Value).Days + 1;
                return 0;
            }
        }
    }
}
